javaparser-core/src/main/java/com/github/javaparser/utils/SourceRoot.java

```diff
@@ -50,6 +46,9 @@
     private JavaParser javaParser = new JavaParser();
 
     public SourceRoot(Path root) {
+        if (Files.isRegularFile(root)) {
```

Comment:
@matozoid I'm not 100% sure if one is the complement of the other (ie, `!Files.isDirectory` == `Files.isRegularFile`) but this is apparently what @arturbosch meant initially. I assumed this by the exception message.